adoc/chapters/cuda_backend.adoc

```diff
@@ -0,0 +1,598 @@
+and `unsampled_image_accessor` classes; `sycl::cuda::texture` and
+`sycl::cuda::surface` respectively.
+
+Below is a table of the `backend_return_t` specializations
+for the SYCL classes which support kernel function interoperability.
+
+[[table.cuda.kernelinterop.nativeobjects]]
+.Types of native backend objects kernel function interoperability
+[width="100%",options="header",cols="20%,20%,20%,40%"]
+|====
+| [code]#SyclType#                                                   | [code]#backend_return_t<backend::cuda, SyclType># | Description
+| [code]#accessor<T, Dims, Mode, target::device>#                    | void * | Convert a SYCL `accessor` to an undecorated raw pointer.
+| [code]#accessor<T, Dims, Mode, target::constant_buffer>#           | void * | Convert a SYCL `accessor` to an undecorated raw pointer.
+| [code]#accessor<T, Dims, Mode, target::local>#                     | void * | Convert a SYCL `accessor` to an undecorated raw pointer.
+| [code]#local_accessor<T, Dims>#                                    | void * | Convert a SYCL `local_accessor` to an undecorated raw pointer.
+| [code]#sampled_image_accessor<T, 1, Mode, image_target::device>#   | sycl::cuda::texture<T, 1> | Convert a SYCL `sampled_image_accessor` to the `sycl::cuda::texture` interoperability type with the same type and dimensions.
+| [code]#sampled_image_accessor<T, 2, Mode, image_target::device>#   | sycl::cuda::texture<T, 2> | Convert a SYCL `sampled_image_accessor` to the `sycl::cuda::texture` interoperability type with the same type and dimensions.
+| [code]#sampled_image_accessor<T, 3, Mode, image_target::device>#   | sycl::cuda::texture<T, 3> | Convert a SYCL `sampled_image_accessor` to the `sycl::cuda::texture` interoperability type with the same type and dimensions.
+| [code]#unsampled_image_accessor<T, 1, Mode, image_target::device># | sycl::cuda::surface<T, 1> | Convert a SYCL `unsampled_image_accessor` to the `sycl::cuda::surface` interoperability type with the same type and dimensions.
+| [code]#unsampled_image_accessor<T, 2, Mode, image_target::device># | sycl::cuda::surface<T, 2> | Convert a SYCL `unsampled_image_accessor` to the `sycl::cuda::surface` interoperability type with the same type and dimensions.
+| [code]#unsampled_image_accessor<T, 3, Mode, image_target::device># | sycl::cuda::surface<T, 3> | Convert a SYCL `unsampled_image_accessor` to the `sycl::cuda::surface` interoperability type with the same type and dimensions.
+|====
+
+[[sec:cuda_support_of_core_features]]
+== CUDA Support of Core SYCL Features
+
+Some core SYCL features require a minimum compute capability for the CUDA
+backend.
+
+[[table.extensionsupport]]
+.CUDA support for Core SYCL API features
+[width="100%",options="header",cols="33%,33%,33%"]
+|====
+| Feature                                   | SYCL Aspect               | Required Compute Capability 
+| [code]#16-bit floating point#             | [code]#aspect::fp16#      | 5.3 or greater
+|====
```

Comment:
I can personally see how it isn't strictly necessary, however IMO it is quite concise and could be useful to have this table since the compute capability requirement information is not always easy to find out and it seems nice for people to know the requirement before they explicitly try to write code calling the aspect. @hdelan what do you think?